scipy/stats/stats.py

```diff
@@ -346,6 +346,26 @@ def _broadcast_shapes_with_dropped_axis(a, b, axis):
     return shp
 
 
+def _convert_symmetric_p_value(pvalue, root_func, alternative):
+    """
+    Convert a one-sided, symmetric P-value to a one or two-sided p-value. root_func
+    describes the source distribution function ('sf' or 'cdf')
+    """
+    if alternative not in ["two-sided", "greater", "less"]:
+        raise ValueError("alternative should be 'less', 'greater' or 'two-sided'")
+
+    if root_func not in ["sf", "cdf"]:
+        raise ValueError("root_func should be 'sf' or 'cdf'")
+
+    op_type = alternative + "_" + root_func
+    if op_type == "greater_sf" or op_type == "less_cdf":
```

Comment:
String operations aren't very efficient in Python, but more importantly, I manipulating strings seems a bit indirect in Python. 

Something more familiar to other developers might be:
```suggestion
    if (alternative, root_func) in {("greater", "sf"), ("less", "cdf")}
```

But I was actually going to suggest some renaming that I think would simplify the logic. It seems to me that regardless of whether the function used to calculate the incoming one-sided *p*-value is `sf` or `1-cdf`, either way it's still a `'greater'` *p*-value. Likewise, whether it's implemented as `cdf` or `1-sf` or something else entirely (because remember, there are some tests for which they don't even call a stats `cdf` or `sf`), it's still a `'less'` *p*-value. So maybe if `in_type` is `'less'` or `'greater'` and `out_type` is `'less'`, `'greater'`, `'two-sided'`, then:
```
if out_type == "two-sided":
    return 2 * np.minimum(pvalue, 1 - pvalue)  # although if pvalues are never arrays, maybe just use `min`?
elif out_type != in_type:
    return 1 - pvalue
else:
    return pvalue
```